datastream/dataset.py

```diff
@@ -105,17 +105,29 @@ def map(self, function: Callable) -> Dataset:
             functions=self.functions + tuple([function]),
         )
 
-    # TODO: handle input better. Only allow masks?
-    def subset(self, indices_or_mask: Union[pd.Series, np.array, List[int]]) -> Dataset:
+    def subset(
+        self, mask: Union[pd.Series, np.array, List[bool]]
+    ) -> Dataset:
         '''
-        Select a subset of the dataset using a ``pd.Series`` mask or indices.
+        Select a subset of the dataset using a boolean mask.
         '''
-        if type(indices_or_mask) is pd.Series:
-            indices_or_mask = np.argwhere(indices_or_mask.values).squeeze(1)
+        if isinstance(mask, list):
+            mask = np.array(mask)
+        elif isinstance(mask, pd.Series):
+            mask = mask.values
 
+        if len(mask.shape) != 1:
+            raise AssertionError('Expected single dimension in mask')
```

Comment:
Could throw https://docs.python.org/3.8/library/exceptions.html#ValueError
(depending if you see the shape as part of the type or not)

Reply:
No opinion on what's better. We can change this later if you prefer ValueError